### PR TITLE
Detect orphaned file locks and unlock

### PR DESCRIPTION
**Pull request by Connorhd**

### CODE REVIEW

File: lib/file_mutex.rb [hunk 1]
```diff
@@ -56,6 +57,10 @@ def lock!(timeout_duration = DEFAULT_TIMEOUT)
     end
     return result
   rescue Errno::EEXIST
+    File.open(@path) do |f|
+      # If we can lock the file the original process must have died - unlock now
```
[the remark this answers is not in the document]
Comment (dliusquare) [reply lib/file_mutex.rb]: Thought more about this implementation and noticed there is a race condition: when first job create the file, the second job fails to create, then open, flock and delete the file.

[the remark this answers is not in the document]
Comment (Connorhd) [reply lib/file_mutex.rb]: I think we can fix by retrying if the other lock fails rather than just ignoring success/failure

Comment (dliusquare) [inline lib/file_mutex.rb]: (not sure if my previous comment got lost, here is it again)
The existing code has another race condition: a successful lock will call unlock twice, one in timeout block and one in lock! block. The second unlock! may accidentally delete the file locked by other process.

**Comment by dliusquare** [comment]
Other solution has been merged. Close this PR.

**Pull request closed without merging**